### ISSUE TITLE
`<ranges>`: could not use std::views::istream

### ISSUE THREAD

**Issue by xiaosa-zhz** [StephanTLavavej/STL#13]
# Basic Info
* [x] I have read the rules for the [Standard Library Modules Bug Bash][bug-bash-url].

* [x] I have pulled and rebuilt the `bug-bash` branch in the last 24 hours.

* My current commit, as printed by `git rev-parse HEAD`, is: 6ddfaf5d6a4b2096dca10ba952c39fb42a2f557d

* This bug is:
rejects-valid

# Command-Line Test Case
My test case that works with classic includes but fails with named modules:
```
c:\STL\STL>type helloworld.cpp
#ifdef USE_CLASSIC_INCLUDES
#include <ranges>
#include <iostream>
#include <algorithm>
#else
import std;
#endif // USE_CLASSIC_INCLUDES

namespace stdr = std::ranges;
namespace stdv = std::views;

int main()
{
    auto input = stdv::istream<int>(std::cin)
        | stdv::take_while([](auto e) { return e != 0; });
    stdr::for_each(input, [](auto e) { std::cout << e << '\n'; });
}
c:\STL\STL>cl /EHsc /nologo /W4 /std:c++latest /MTd /Od helloworld.cpp /DUSE_CLASSIC_INCLUDES && helloworld.exe
helloworld.cpp
1 2 3 0
1
2
3

c:\STL\STL>cl /EHsc /nologo /W4 /std:c++latest /MTd /Od helloworld.cpp std.obj
helloworld.cpp
helloworld.obj : error LNK2019: 无法解析的外部符号 "struct _Istream_fn<int>::std::ranges::_Istream_fn<int> const std::ranges::views::istream<int>" (??$istream@H@views@ranges@std@@3U?$_Istream_fn@H@123@B::<!std>)，函数 _main 中引用了该符号
helloworld.exe : fatal error LNK1120: 1 个无法解析的外部命令
```

# Additional Context
Do I get the format right?

[bug-bash-url]: https://github.com/StephanTLavavej/STL/wiki/Standard-Library-Modules-Bug-Bash

**Comment by StephanTLavavej** [comment]
The format is great, thanks for the clear test case! I agree this is a rejects-valid bug and will investigate. :heart_eyes_cat:

**Comment by StephanTLavavej** [comment]
Error message in English:

```
helloworld.obj : error LNK2019: unresolved external symbol "struct _Istream_fn<int>::std::ranges::_Istream_fn<int> const std::ranges::views::istream<int>" (??$istream@H@views@ranges@std@@3U?$_Istream_fn@H@123@B::<!std>) referenced in function main
helloworld.exe : fatal error LNK1120: 1 unresolved externals
```

`std::ranges::views::istream` is exported, so this is a compiler bug:

https://github.com/StephanTLavavej/STL/blob/6ddfaf5d6a4b2096dca10ba952c39fb42a2f557d/stl/inc/ranges#L1432-L1433

I'll reduce and report this to the compiler team tomorrow!

**Comment by StephanTLavavej** [comment]
Reduced and reported as internal [VSO-1581022](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1581022) "Standard Library Modules: `inline constexpr` variables for `ranges::views` emit `error LNK2019: unresolved external symbol`":

```
C:\Temp>type meow.ixx
```
```cpp
export module meow;

template <typename T>
struct IstreamFn {
    constexpr void operator()() const {}
};

export template <typename T>
inline constexpr IstreamFn<T> istream;
```
```
C:\Temp>type user.cpp
```
```cpp
import meow;

int main() {
    istream<double>();
}
```
```
C:\Temp>cl /EHsc /nologo /W4 /std:c++latest /Feoutput.exe meow.ixx user.cpp
meow.ixx
user.cpp
Generating Code...
user.obj : error LNK2019: unresolved external symbol "struct IstreamFn<double> const istream<double>" (??$istream@N@@3U?$IstreamFn@N@@B::<!meow>) referenced in function main
output.exe : fatal error LNK1120: 1 unresolved externals
```

**Comment by StephanTLavavej** [comment]
This works with the current development build of the compiler (prod/fe amd64chk), so I believe that this fix will appear in 17.4 Preview 2 or earlier.

**Comment by StephanTLavavej** [comment]
@cdacamar confirmed that this was fixed in 17.4 Preview 1. :tada:

**Comment by StephanTLavavej** [comment]
I've verified that the compiler fix has shipped in 17.4 Preview 1 and that this test case now compiles successfully.
